File: v1/src/main/java/com/api/v1/borrower/controllers/BorrowerSelfRegistrationController.java
package com.api.v1.borrower.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import com.api.v1.borrower.dtos.NewBorrowerRequestDto;
import com.api.v1.borrower.dtos.BorrowerResponseDto;
import com.api.v1.borrower.services.BorrowerSelfRegistrationService;

import jakarta.validation.Valid;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/borrowers")
public class BorrowerSelfRegistrationController {

    @Autowired
    private BorrowerSelfRegistrationService service;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Mono<BorrowerResponseDto> selfRegister(@Valid @RequestBody NewBorrowerRequestDto request) {
        return service.selfRegister(request);
    }

}

File: koltlin/v1/src/main/kotlin/com/api/v1/borrower/controllers/BorrowerSelfRegistrationController.kt
package com.api.v1.borrower.controllers

import com.api.v1.borrower.helpers.BorrowerResponseDto
import com.api.v1.borrower.helpers.NewBorrowerRequestDto
import com.api.v1.borrower.services.BorrowerSelfRegistrationService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Mono

@RestController
@RequestMapping("api/v1/borrowers")
class BorrowerSelfRegistrationController {

    private val service: BorrowerSelfRegistrationService

    constructor(service: BorrowerSelfRegistrationService) {
        this.service = service
    }

    @PostMapping
    @ResponseStatus(value = HttpStatus.CREATED)
    fun selfRegister(@Valid @RequestBody request: NewBorrowerRequestDto): Mono<BorrowerResponseDto> {
        return service.selfRegister(request)
    }

}